src/world_renderer.rs

```diff
@@ -526,10 +522,15 @@ impl WorldRenderer {
             MouseButton::Left => {
                 if self.player.is_selecting_cube() {
                     self.player.toggle_state(MotionState::LeftClick, state.is_pressed());
-                } else if let Some(mut attack) = self.entity_manager.attack(self.player.position().pos(), self.player.direction()) {
-                    // Forward the attack to the server
-                    attack.set_strength(self.items.attack_strength());
-                    self.proxy.lock().unwrap().on_new_attack(attack);
+                } else if state.is_pressed() {
+                    if let Some(mut attack) = self.entity_manager.attack(self.player.position().pos(), self.player.direction()) {
+                        if !attack.entity_kind().is_player() {
+                            self.entity_manager.remove_entity(attack.victim_id());
+                        }
+                        // Forward the attack to the server
+                        attack.set_strength(self.items.attack_strength());
+                        self.proxy.lock().unwrap().on_new_attack(attack);
+                    }
```

Comment:
This is wrong. It's not the role of the world renderer to kill monsters !!! The manager simply forwards to the server the attack and the server forwards the consequence.